### PR TITLE
feat(core/dfn): add support for task source types

### PR DESCRIPTION
**Pull request by marcoscaceres**
This automatically detects and exports "task source" definitions, which will make them easier for find and reference.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@saschanaz, is it ok if ping you for the occasional review? @sidvishnoi is taking a break. 

I won't bother you with refactors or small code changes.

**Review by saschanaz — APPROVED** [review]
LGTM, is this something supported by specref/xref?

File: src/core/dfn.js [hunk 1]
```diff
@@ -108,6 +114,13 @@ function computeType(dfn, linkingText) {
...
+function processAsTaskSource(dfn) {
+  // Task sources are exported by default.
+  if (!dfn.matches(".no-export, [data-noexport]")) {
+    dfn.dataset.export = "";
+  }
+}
+
 // Deal with export/no export
 function computeExport(dfn) {
   switch (true) {
```
Comment (saschanaz) [inline src/core/dfn.js:126]: I'm not sure how I feel about this pattern 🤔😂

Comment (marcoscaceres) [reply src/core/dfn.js:126]: At first, I was like... that's weird/not right... but now I really love how it split up the code into different cases. 

Eventually, once ECMAScript gets pattern matching, we can switch them all over to that:
https://github.com/tc39/proposal-pattern-matching

**Pull request merged**